cranelift/codegen/src/dominator_tree.rs

```diff
@@ -482,7 +482,42 @@ impl DominatorTreePreorder {
     pub fn dominates(&self, a: Block, b: Block) -> bool {
         let na = &self.nodes[a];
         let nb = &self.nodes[b];
-        na.pre_number <= nb.pre_number && na.pre_max >= nb.pre_max
+        // debug_assert!(na.pre_number != 0);
+        // debug_assert!(nb.pre_number != 0);
```

Comment:
Does the test suite pass with these assertions uncommented? You're testing `nb` explicitly for being unreachable on line 491, so I'm guessing that at least that one doesn't pass?

Reply:
hi Elliot, yes one of the test suite failed with this assertion.  Here is the summary of the test suite through local ci: 
```
failures:

---- dominator_tree::tests::unreachable_node stdout ----
thread 'dominator_tree::tests::unreachable_node' panicked at cranelift/codegen/src/dominator_tree.rs:486:9:
assertion failed: nb.pre_number != 0


failures:
    dominator_tree::tests::unreachable_node

test result: FAILED. 187 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.21s

error: test failed, to rerun pass `-p cranelift-codegen --lib`
```